src/CowSdk.ts

```diff
@@ -1,13 +1,58 @@
+import { JsonRpcProvider } from '@ethersproject/providers'
 import { version as SDK_VERSION } from '../package.json'
+import { CowApi } from './api'
+import { DEFAULT_APP_DATA_HASH } from './constants'
+import { SupportedChainId as ChainId } from './constants/chains'
 import { validateAppDataDocument } from './utils/appData'
-import { signOrder, signOrderCancellation } from './utils/sign'
+import { SdkError } from './utils/common'
+import { signOrder, signOrderCancellation, UnsignedOrder } from './utils/sign'
 
-export class CowSdk {
+export interface SdkContext {
+  appDataHash?: string
+  isDevEnvironment?: boolean
+  provider?: JsonRpcProvider
+}
+
+export class CowSdk<T extends ChainId> {
   static version = SDK_VERSION
+  chainId: T
+  appDataHash: string
+  provider?: JsonRpcProvider
+  api: CowApi<T>
+
+  constructor(
+    chainId: T,
+    { appDataHash, isDevEnvironment, provider }: SdkContext = {
```

Comment:
move to DEFAULT_SDK_CONTEXT the object? 

Also, destructure in the body of the constructor, to keep the function signature smaller and easier to read?